File: dotnet/DesignPatterns/BehavioralDesignPatterns/TemplateMethodDesignPattern/TemplateMethodDesignPattern.Example1/DeveloperRecruitmentProcess.cs
namespace TemplateMethodDesignPattern.Example1;

public class DeveloperRecruitmentProcess : RecruitmentProcess
{
    protected override void TechnicalInterview()
    {
        Console.WriteLine("Candidate was tested for coding skills and system design via online coding assessment");
    }

    protected override void InterviewWithManager()
    {
        Console.WriteLine("Candidate interviewed with the Tech Lead and Engineering Manager");
    }

    protected override void NotifyCandidate()
    {
        Console.WriteLine("We sincerely appreciate the time and effort you invested during the selection process.\nAfter thoughtful evaluation, we have decided not to move forward with your application.");
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/TemplateMethodDesignPattern/TemplateMethodDesignPattern.Example1/src/DeveloperRecruitmentProcess.ts
import { RecruitmentProcess } from "./RecruitmentProcess";

export class DeveloperRecruitmentProcess extends RecruitmentProcess {
  protected technicalInterview(): void {
    console.log(
      "Candidate was tested for coding skills and system design via online coding assessment"
    );
  }

  protected interviewWithManager(): void {
    console.log("Candidate interviewed with the Tech Lead and Engineering Manager");
  }

  protected notifyCandidate(): void {
    console.log(
      "We sincerely appreciate the time and effort you invested during the selection process.\n" +
        "After thoughtful evaluation, we have decided not to move forward with your application."
    );
  }
}
